go/vt/vtgate/executor.go

```diff
@@ -1420,7 +1420,8 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	vcursor.SetIgnoreMaxMemoryRows(ignoreMaxMemoryRows)
 
 	// Normalize if possible and retry.
-	if (e.normalize && sqlparser.CanNormalize(stmt)) || sqlparser.MustRewriteAST(stmt, qo.getSelectLimit() > 0) {
+	// We do not normalize already prepared statements, by default
+	if (e.normalize && (*normalizePrepStmts || len(bindVars) == 0) && sqlparser.CanNormalize(stmt)) || sqlparser.MustRewriteAST(stmt, qo.getSelectLimit() > 0) {
```

Comment:
Can you triple check that `len(bindVars) == 0` is the right logic here? I see you went back and forth on it in your Git history. :p

...We're saying that a prepared statement cannot have bind variables. Is that right?

Reply:
We are saying that prepared statements *will* have > 0 bind variables, and thus are eligible for normalization (or, if normalization is being forced).

Reply:
OK I'm a bit lost here...  The title of the PR is "Bypass normalization of already prepared statements", which means that prepared statements should _not_ be eligible for normalization.

Reply:
missed a *not* in my comment above